### PR TITLE
Add model for showing and importing vCalendar and iCalendar events from file

### PR DESCRIPTION
**Pull request by pgerdt**
The contents of a vCalendar or iCalendar file are read to memory, and a model is populated with the event data from the file. Incidences that are not events are discarded. Events can then be imported to a notebook.

### CODE REVIEW

File: src/calendarutils.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    return result;
+}
+
+NemoCalendarData::EventOccurrence NemoCalendarUtils::getNextOccurrence(const KCalCore::Event::Ptr &event,
+                                                           const QDateTime &start)
+{
+    NemoCalendarData::EventOccurrence occurrence;
+    if (event) {
+            QDateTime dtStart = event->dtStart().toLocalZone().dateTime();
+            QDateTime dtEnd = event->dtEnd().toLocalZone().dateTime();
```
Comment (pvuorela) [inline src/calendarutils.cpp]: Extra indentation.

File: src/calendarimportmodel.cpp [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+void NemoCalendarImportModel::setFileName(const QString &fileName)
+{
+    if (mFileName == fileName)
+        return;
+
+    mFileName = fileName;
+    emit fileNameChanged();
+    if (!mFileName.isEmpty())
+        importToMemory(fileName);
```
Comment (pvuorela) [inline src/calendarimportmodel.cpp]: Should setting empty file name also clear current events from memory?

File: src/calendarimportmodel.cpp [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    case LocationRole:
+        return event->location();
+    case UidRole:
+        return event->uid();
+    default:
+        return QVariant();
+    }
+}
+
+bool NemoCalendarImportModel::importToNotebook(const QString notebookUid)
```
Comment (pvuorela) [inline src/calendarimportmodel.cpp]: Should probably be const reference.

File: src/calendarimportmodel.cpp [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        mEventList.clear();
+
+    beginResetModel();
+    KCalCore::MemoryCalendar::Ptr cal(new KCalCore::MemoryCalendar(KDateTime::Spec::LocalZone()));
+    importFromFile(fileName, cal);
+    KCalCore::Incidence::List incidenceList = cal->incidences();
+    for (int i = 0; i < incidenceList.size(); i++) {
+        KCalCore::Incidence::Ptr incidence = incidenceList.at(i);
+        if (incidence->type() == KCalCore::IncidenceBase::TypeEvent)
+            mEventList.append(incidence.staticCast<KCalCore::Event>());
```
Comment (pvuorela) [inline src/calendarimportmodel.cpp]: To be noted that exceptions, i.e. instances having recurrence id, on recurring events will show up as separate incidences on this model.

File: src/calendarutils.h [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#include "calendardata.h"
+
+// kCalCore
+#include <event.h>
+
+class NemoCalendarUtils
+{
+public:
+    NemoCalendarUtils();
+    ~NemoCalendarUtils();
```
Comment (pvuorela) [inline src/calendarutils.h]: As mentioned, the class part is quite unnecessary and namespace would suffice.

**Comment by pgerdt** [comment]
Updated PR:
- Changed class NemoCalendarUtils into a namespace
- Clear model when filename is empty (https://github.com/nemomobile/nemo-qml-plugin-calendar/pull/52/files#diff-6a5cb3ed7dc535bdc9481bbd84566208R78)
- Use const reference, indentation fixes

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**